chain/network/src/tests/util.rs

```diff
@@ -0,0 +1,47 @@
+pub type Instant = chrono::DateTime<chrono::Utc>;
+pub type Duration = chrono::Duration;
+
+pub trait Clock {
+    fn now(&self) -> Instant;
+}
```

Comment:
Yessss, :heart_eyes: this is exactly the thing we need. This is just great.

Note that we have an attempt at this API at https://github.com/near/nearcore/blob/master/core/primitives/src/time.rs, but that was severely messed up by our desire to ship something, rather than do a proper, principled thing. 

Only two comments here: 

* I wouldn't call `DateTime<chrono::Utc>` an `Instant`. `Instant` has "monotonic" connotation, which is wrong for `DateTime`. Id' just call it `Time`. 
* I'd hide the `trait` inside, and make the public type a `struct Clock;`. The fact that the caller needs to type `dyn` leaks impl details

Reply:
Addndum: `Clock::real / Clock::fake` also looks cute and allows to keep impls private.

Reply:
Ack. The monotonic connotation was intended. If anything I'd prefer to replace DateTime with sth more suitable for an output of a monotonic clock. What kind of timestamp type would be suitable for that, while at the same time satisfying the requirements of our codebase in the long run (I see chrono::DateTime used quite a lot)?.

Reply:
I'll make the clock support both monotonic time and unix time.